v2/result.go

```diff
@@ -26,3 +26,22 @@ type HealthResult struct {
 	Ok            bool          `json:"ok"`
 	Severity      uint8         `json:"severity,omitempty"`
 }
+
+func ComputeOverallStatus(result *HealthResult) bool {
```

Comment:
I had an inner debate, I actually wan't to keep this structs simple as Java POJOs, and the main reason I didn't attached the method to the HealthResult struct is because the Ok attribute is already exposed public, if I add a method getOverallStatus() it's and information in 2 places, which are not syncronized, there is the risk that directly get's the Ok attribute value because it's public and actually the value is wrong because it's not computed.
If I'd add the method automatically set the Ok that you have to call a couple of self methods on a POJO so that it's values would be relevant, it's seems nasty and error prone to me.
I'd rather have a POJO which I manipulate from the exterior as I want, passing it as an parameter, I always know the values are what I want...